crates/solver-types/src/api.rs

```diff
@@ -303,34 +307,230 @@ impl TryFrom<(&Quote, &str, &str)> for IntentRequest {
+		let input_token_u256 = U256::from_str_radix(&token_id_hex, 16)
+			.map_err(|e| format!("Failed to parse TOKEN_ID: {}", e))?;
+
+		let inputs = vec![[input_token_u256, input_amount]];
+
+		// Extract outputs from mandate
+		let outputs_value = mandate.get("outputs").ok_or("Missing outputs in mandate")?;
+		let mut sol_outputs = Vec::new();
+
+		if let Some(outputs_array) = outputs_value.as_array() {
+			for output_item in outputs_array {
+				if let Some(output_obj) = output_item.as_object() {
+					let chain_id = output_obj
+						.get("chainId")
+						.and_then(|c| {
+							if let Some(s) = c.as_str() {
+								s.parse::<u64>().ok()
+							} else {
+								c.as_u64()
+							}
+						})
+						.unwrap_or(0);
+					let amount_str = output_obj
+						.get("amount")
+						.and_then(|a| a.as_str())
+						.unwrap_or("0");
+					let token_str = output_obj.get("token").and_then(|t| t.as_str()).unwrap();
+					let recipient_str = output_obj
+						.get("recipient")
+						.and_then(|r| r.as_str())
+						.unwrap();
+					let oracle_str = output_obj.get("oracle").and_then(|o| o.as_str()).unwrap();
+					let settler_str = output_obj.get("settler").and_then(|s| s.as_str()).unwrap();
+
+					if let Ok(amount) = U256::from_str_radix(amount_str, 10) {
+						let token_bytes = parse_bytes32_from_hex(token_str).unwrap_or([0u8; 32]);
+						let recipient_bytes =
+							parse_bytes32_from_hex(recipient_str).unwrap_or([0u8; 32]);
+						let oracle_bytes = parse_bytes32_from_hex(oracle_str).unwrap_or([0u8; 32]);
+						let settler_bytes =
+							parse_bytes32_from_hex(settler_str).unwrap_or([0u8; 32]);
+
+						sol_outputs.push(SolMandateOutput {
+							oracle: oracle_bytes.into(),
+							settler: settler_bytes.into(),
+							chainId: U256::from(chain_id),
+							token: token_bytes.into(),
+							amount,
+							recipient: recipient_bytes.into(),
+							call: Vec::new().into(),
+							context: Vec::new().into(),
+						});
+					}
+				}
+			}
+		}
+
+		Ok(StandardOrder {
+			user: user_address,
+			nonce,
+			originChainId: origin_chain_id,
+			expires,
+			fillDeadline: fill_deadline,
+			inputOracle: input_oracle,
+			inputs,
+			outputs: sol_outputs,
+		})
+	}
+
+	/// Main conversion function from EIP-7683 quote to IntentRequest
 	fn from_eip7683_quote(
 		quote: &Quote,
 		signature: &str,
 	) -> Result<Self, Box<dyn std::error::Error>> {
 		use crate::standards::eip7683::interfaces::StandardOrder;
 		use alloy_primitives::Bytes;
 		use alloy_sol_types::SolType;
-		use std::convert::TryFrom;
 
-		let sol_order = StandardOrder::try_from(quote)?;
+		// Detect order type from EIP-712 data
+		let (eip712_data, primary_type) = Self::extract_eip712_data(quote)?;
+
+		let sol_order = if primary_type == "BatchCompact" {
+			// Handle BatchCompact (ResourceLock) orders
+			Self::handle_batch_compact_order(quote, eip712_data)?
+		} else {
+			// Handle Permit2/EIP3009 orders using existing implementation
+			StandardOrder::try_from(quote)?
+		};
```

Comment:
What is handle_batch_compact_order doing structurally different than just try_from here?